### PR TITLE
480 remove article continuous scroll

### PR DESCRIPTION
**Pull request by Rowansdabomb**

### CODE REVIEW

**Comment by Rowansdabomb** [comment]
PR for issues 466, 472, 480

**Review by JamieRL — APPROVED** [review]
Reviewed together with @Rowansdabomb

File: ubyssey/helpers.py [hunk 1]
```diff
@@ -31,6 +31,49 @@ def get_reading_time(article):
         reading_time = word_count / words_per_min
         return reading_time
 
+    @staticmethod
+    def insert_ads(article, article_type = 'desktop'):
```
Comment (psiemens) [inline ubyssey/helpers.py]: Remove extra spaces:

```py
def insert_ads(article, article_type='desktop'):
```

File: ubyssey/helpers.py [hunk 2]
```diff
@@ -31,6 +31,49 @@ def get_reading_time(article):
...
+        """ inject upto 5 ads evenly throughout the article content
```
Comment (psiemens) [inline ubyssey/helpers.py]: Capitalize and punctuate:

```py
"""Inject up to 5 ads evenly throughout the article content. 
Ads can not inject directly beneath headers.""

File: ubyssey/helpers.py [hunk 3]
```diff
@@ -31,6 +31,49 @@ def get_reading_time(article):
...
+        ad = {
+            unicode('type'): unicode('ad'),
+            unicode('data'): unicode(article_type)
+        }
+
+        paragraph_count = 1
+        
+        for block in article.content:
+            if block['type'] == 'paragraph':
+                paragraph_count += 1
```
Comment (psiemens) [inline ubyssey/helpers.py]: Here's a one-liner if you want to reduce code:

```py
paragraph_count = len(filter(lambda b: b['type'] == 'paragraph', article.content))

File: ubyssey/helpers.py [hunk 4]
```diff
@@ -31,6 +31,49 @@ def get_reading_time(article):
...
+        
+        numberOfAds = 1
```
Comment (psiemens) [inline ubyssey/helpers.py]: Use underscores instead of camelcase

File: ubyssey/helpers.py [hunk 5]
```diff
@@ -149,6 +192,15 @@ def get_topic(topic_name):
             topic__name=topic_name
         )
 
+    @staticmethod
+    def is_explicit(article):
+        explicit_tags = ['sex', 'Sex', 'explicit', 'Explicit']
+        tags = article.tags.all().values_list('name', flat=True)
+        for tag in tags:
+            if tag in explicit_tags:
```
Comment (psiemens) [inline ubyssey/helpers.py]: You can do this:

```py
explicit_tags = ['sex', 'explicit']
# ...
if tag.lower() in explicit_tags:

File: ubyssey/helpers.py [hunk 6]
```diff
@@ -31,6 +31,49 @@ def get_reading_time(article):
         reading_time = word_count / words_per_min
         return reading_time
 
+    @staticmethod
+    def insert_ads(article, article_type = 'desktop'):
+        """ inject upto 5 ads evenly throughout the article content
+            ads can not inject directly beneath headers """
+        ad = {
+            unicode('type'): unicode('ad'),
+            unicode('data'): unicode(article_type)
```
Comment (psiemens) [inline ubyssey/helpers.py]: Why the unicode wrapping?

File: ubyssey/static/src/js/components/ArticlePreview.jsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
+import React, { Component } from 'react'
+import humanizeDateTime from '../modules/Dates'
+
+class ArticlePreview extends Component{
+  constructor(props) {
+    super(props)
+  }
```
Comment (psiemens) [inline ubyssey/static/src/js/components/ArticlePreview.jsx]: You can take this method out

File: ubyssey/static/src/js/components/ArticlePreview.jsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  
+  goToArticle() {
+    window.location = this.props.url
+  }
```
Comment (psiemens) [inline ubyssey/static/src/js/components/ArticlePreview.jsx]: Is this method still needed?

File: ubyssey/views/main.py [hunk 9]
```diff
@@ -84,10 +86,19 @@ def article(self, request, section=None, slug=None):
...
+        user_agent = get_user_agent(request)
+        if user_agent.is_mobile:
+            article_type = 'mobile'
+
         ref = request.GET.get('ref', None)
         dur = request.GET.get('dur', None)
 
-        authors_json_name = json.dumps([a.person.full_name for a in article.authors.all()])
+        if not ArticleHelper.is_explicit(article):
+            article = ArticleHelper.insert_ads(article, article_type)
```
Comment (psiemens) [inline ubyssey/views/main.py]: I'd prefer if this was called like this:

```py
article.content = ArticleHelper.insert_ads(article.content, article_type)

File: ubyssey/views/main.py [hunk 10]
```diff
@@ -105,19 +118,28 @@ def article(self, request, section=None, slug=None):
 
     def article_ajax(self, request, pk=None):
```
Comment (psiemens) [inline ubyssey/views/main.py:121]: Do we even need this route anymore?

**Review by psiemens — CHANGES_REQUESTED** [review]
I'm wondering if it makes more sense to insert the suggested articles as plain HTML rather than React. Since the mobile layout will be using a separate carousel, I'm not sure it's necessary to use React in the desktop view. Thoughts? @JamieRL @Rowansdabomb

File: ubyssey/helpers.py (hunk 5 above)
Comment (JamieRL) [reply ubyssey/helpers.py]: nice

**Pull request merged**